src/deploy/functions/runtimes/discovery/v1alpha1.ts

```diff
@@ -250,16 +250,15 @@ function parseEndpoints(
       "secretEnvironmentVariables",
       "secretEnvironmentVariables",
       (senvs: ManifestEndpoint["secretEnvironmentVariables"]) => {
-        if (senvs && senvs.length > 0) {
-          ep.secretEnvironmentVariables = [];
-          for (const { key, secret } of senvs) {
-            ep.secretEnvironmentVariables.push({
-              key,
-              secret: secret || key, // if secret is undefined, assume env var key == secret name
-              projectId: project,
-            });
-          }
+        const secretEnvironmentVariables: backend.SecretEnvVar[] = [];
+        for (const { key, secret } of senvs!) {
+          secretEnvironmentVariables.push({
+            key,
+            secret: secret || key, // if secret is undefined, assume env var key == secret name
+            projectId: project,
+          });
         }
+        return secretEnvironmentVariables;
```

Comment:
Can there be a test written for this case? Running coverage for the test suite for this file tells me this block of code isn't exercised :)